app/src/test/java/http_server/ResponseFormatterTest.java

```diff
@@ -0,0 +1,39 @@
+package http_server;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.startsWith;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ResponseFormatterTest {
+
+  @Test
+  public void testBuildResponse() {
+
+    Response response = mock(Response.class);
+
+    Map<String, String> responseHeaders = new HashMap<>();
+    responseHeaders.put("Content-Type", "text/plain");
+    responseHeaders.put("Server", "Test Server");
+
+    when(response.getResponseHeaders()).thenReturn(responseHeaders);
+    when(response.getResponseBody()).thenReturn("Hello, world!");
+
+    ResponseFormatter formattedResponse = new ResponseFormatter();
```

Comment:
similar to the other feedback, i'd just call those responseFormatter.  when you instantiate the class, nothing has been "formatted" yet. again, it's a small detail but it communicates what the variable actually contains.